### PR TITLE
feat(processors.timestamp): Introduce timestamp processor

### LINKED ISSUE #10524: Support for parsing custom/multiple time fields in data parsers
I've been dealing with a rather simple JSON data which shows health metrics from multiple mobile devices where battery level and last seen date is reported. I'm parsing it with inputs.file and I've used timestamp_key and timestamp_format to get the "last seen" field key and it is correctly stored in the time field for the metric. However, if someone wants to keep the time of the actual data log and store "last seen" in a separate field in unix timestamp format, it doesn't seem possible? Or, for example, if you have several date fields, there is no way to convert them all to unix timestamp?

### Proposal:
timestamp_key to support multiple fields and formats per-field or processors.converter to support date parsing

### Current behavior:
timestamp_key supports only one field and stores it in "time", no conversion is possible with processors.converter either

### Desired behavior:
add support either in inputs directly, or in processors.converter where dates to be converted to timestamp format

### Use case:
Pretty much any JSON that has more than one timestamp field that needs to be converted to proper unix timestamp

### PR DESCRIPTION
**Pull request by powersj**
## Summary
The following introduces a new timestamp processor. This allows a user to specify a field which contains a timestamp and convert it from one format to another. Traditionally telegraf assumes data has a single timestamp and uses that as the timestamp for the metric itself. However, if a user's data contains additional timestamps that they also wish to parse into a different format, there was no existing method for that.

The date processor exists today to take a metric's timestamp and add a tag or field with a subset of that timestamp. Trying to adapt that config to do both features made for a complex configuration that was not straightforward and error prone.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #10524

### CODE REVIEW

File: plugins/processors/timestamp/timestamp.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+			default:
+				stringField, err := internal.ToString(field)
+				if err != nil {
+					continue
+				}
+				timestamp, err = time.ParseInLocation(t.SourceFormat, stringField, t.location)
+				if err != nil {
+					continue
+				}
+			}
```
Comment (srebhan) [inline plugins/processors/timestamp/timestamp.go:96]: Any reason you are not using [`internal.ParseTimestamp()`](https://github.com/influxdata/telegraf/blob/aa1091aba8e6176c2786aef64ae21e8bf66565fe/internal/internal.go#L255) here?

File: plugins/processors/timestamp/timestamp.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		}
+	}
+
+	return in
+}
+
+func init() {
+	processors.Add("timestamp", func() telegraf.Processor {
+		return &Timestamp{
+			SourceTimezone: "UTC",
```
Comment (srebhan) [inline plugins/processors/timestamp/timestamp.go:119]: I would set this in `Init()`...

**Review by srebhan — COMMENTED** [review]
Thanks @powersj! Just a few comments/questions from my side...

File: plugins/processors/timestamp/timestamp.go (hunk 1 above)
Comment (powersj) [reply plugins/processors/timestamp/timestamp.go:96]: thank you for the reminder :)

File: plugins/processors/timestamp/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  destination_timestamp_format = ""
+
+  ## Timestamp Timezone
+  ## Source timestamp timezone. If not set, assumed to be in UTC.
+  ## Options are as follows:
+  ##   1. UTC                 -- or unspecified will return timestamp in UTC
+  ##   2. Local               -- interpret based on machine localtime
+  ##   3. "America/New_York"  -- Unix TZ values like those found in
+  ##        https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+  # source_timestamp_timezone = ""
```
[the remark this answers is not in the document]
Comment (powersj) [reply plugins/processors/timestamp/README.md:44]: sure!

File: plugins/processors/timestamp/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+## Configuration
+
+```toml @sample.conf
+# Convert a timestamp field to other timestamp format
+[[processors.timestamp]]
+  ## Timestamp key to convert
+  ## Specify the field name that contains the timestamp to convert. The result
+  ## will replace the current field value.
+  source_timestamp_field = ""
```
[the remark this answers is not in the document]
Comment (powersj) [reply plugins/processors/timestamp/README.md:23]: done

**Comment by powersj** [comment]
@srebhan if you could take another look I would apprecaite it. Not sure about the destination timezone handling and if I could re-use anything we already have?

**Review by srebhan — APPROVED** [review]
Awesome! Thanks @powersj!

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**